src/Arcus.Messaging.Tests.Integration/MessagePump/ServiceBusMessagePumpTests.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Arcus.EventGrid.Parsers;
+using Arcus.EventGrid.Testing.Infrastructure.Hosts.ServiceBus;
+using Arcus.Messaging.ServiceBus.Core.Extensions;
+using Arcus.Messaging.Tests.Core.Events.v1;
+using Arcus.Messaging.Tests.Core.Generators;
+using Microsoft.Azure.EventGrid.Models;
+using Microsoft.Azure.ServiceBus;
+using Microsoft.Azure.ServiceBus.Core;
+using Microsoft.Extensions.Configuration;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Arcus.Messaging.Tests.Integration.MessagePump
+{
+    [Trait("Category", "Integration")]
+    public class ServiceBusMessagePumpTests : IntegrationTest, IAsyncLifetime
+    {
+        private const string QueueConnectionStringKey = "Arcus:ServiceBus:ConnectionStringWithQueue";
+        private const string TopicConnectionStringKey = "Arcus:ServiceBus:ConnectionStringWithTopic";
+
+        public static IEnumerable<object[]> Encodings
+        {
+            get
+            {
+                yield return new object[] { Encoding.UTF8 };
+                yield return new object[] { Encoding.UTF7 };
+                yield return new object[] { Encoding.UTF32 };
+                yield return new object[] { Encoding.ASCII };
+                yield return new object[] { Encoding.Unicode };
+                yield return new object[] { Encoding.BigEndianUnicode };
+            }
+        }
+
+        private ServiceBusEventConsumerHost _serviceBusEventConsumerHost;
+
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="ServiceBusMessagePumpTests" /> class.
+        /// </summary>
+        public ServiceBusMessagePumpTests(ITestOutputHelper testOutput) : base(testOutput)
+        {
+        }
+
+        [Theory]
+        [MemberData(nameof(Encodings))]
+        public async Task ServiceBusQueueMessagePump_PublishServiceBusMessage_MessageSuccessfullyProcessed(Encoding messageEncoding)
```

Comment:
It was by changing the message pump, that I figured that we may need to add some integration test scenario's here to simulate other message states.
* Start/Stop
* Abadon message
* DeadLettered message
...

Reply:
Wouldn't that be a bit of testing the SDK then? 🤔 

Could be possible but maybe via a follow-up PR or prefer to do it now?

Reply:
Was just thinking, because a problem I found in the stopping of the job was found during my review and not by a test. But could be me.
Is fine, I guess. Maybe we could come up with more E2E tests where the full capability of this is tested with other services.

Reply:
> Was just thinking, because a problem I found in the stopping of the job was found during my review and not by a test. But could be me.
What are you referring to here? 🤔 

Added #44 as well.

Reply:
Was that issue when we forgot to `return` after the message was abandonned and we would still process further.
I think that was something that would need to come up in a test.